app/Models/Course.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\SoftDeletes;
+
+class Course extends Model
+{
+    use HasFactory, SoftDeletes;
+
+    protected $table = "courses";
+    protected $dates = "deleted_at";
+
+    public function lesson()
+    {
+        return $this->hasMany(Lesson::class, 'course_id');
+    }
+
+    public function tag()
+    {
+        return $this->belongsToMany(Tag::class, 'course_tags', 'course_id', 'tag_id');
+    }
+
+    public function user()
+    {
+        return $this->belongsToMany(User::class, 'course_users', 'course_id', 'user_id');
+    }
+
+    public function reviews()
+    {
+        return $this->morphMany(Review::class, 'reviewtable');
```

Comment:
quan hệ này là gì nhỉ?

Reply:
Em sử dụng quan hệ 1 nhiều đa hình
https://laravel.com/docs/8.x/eloquent-relationships#one-to-many-polymorphic-relations